### ISSUE TITLE
Use `torch.nn.CrossEntropyLoss` instead of BCE with one-hot vectors

### ISSUE THREAD

**Issue by ma595** [Cambridge-ICCS/practical-ml-with-pytorch#83]
Simplify loss function and therefore avoid using one-hot vectors. This unnecessarily complicates the approach.
